frontend/src/pages/MetricCreateEdit.jsx

```diff
@@ -0,0 +1,181 @@
+  const metricDescRef = useRef(null);
+    
+
+  const postMetric = async (e) => {
+    
+    const metricName = metricNameRef.current;
+    const metricType = metricTypeRef.current;
+    const metricDesc = metricDescRef.current; 
+
+    let formData = {
+        name: metricName.value,
+        type: metricType.value,
+        area: tags,
+        description: metricDesc.value
+    }
+
+    var data = await fetch('http://127.0.0.1:8000/api/v1/metrics', {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(formData),
+    })
+    .then((response) => response.json())
+    var responseData = {
+        message: data.message,
+        object: data.object
+    }
+    console.log(responseData);
+  }
+  
+  const editMetric = async () => {
+    
+    const metricName = metricNameRef.current;
+    const metricType = metricTypeRef.current;
+    const metricDesc = metricDescRef.current; 
+
+    let formData = {
+        name: metricName.value,
+        type: metricType.value,
+        area: tags,
+        description: metricDesc.value
+    }
+
+    var data = await fetch(`http://127.0.0.1:8000/api/v1/metrics/${metric_id}`, {
+        method: 'PUT',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(formData),
+    })
+    .then((response) => response.json())
+    var responseData = {
+        message: data.message,
+        object: data
+    }
+    console.log(responseData)
+  }
+
+  const getMetricById = async (metric_id) => {
+    try{
+        var data = await fetch(`http://127.0.0.1:8000/api/v1/metrics/${metric_id}`)
+        .then((response) => response.json());
+        
+        if( data.message == "Not Found") {
+            // replace path history to a 404 not found page
+            navigate('/404');
+        }
+
+        var metric_data = data;
+        console.log(metric_data);
+        
+        const metricName = metricNameRef.current;
+        const metricType = metricTypeRef.current;
+        const metricDesc = metricDescRef.current; 
+
+        metricName.value = metric_data.name;
+        metricType.value = metric_data.type;
+        metricDesc.value = metric_data.description;
+
+        setTags([...metric_data.area]);
+
+    } catch (error) {
+        console.log(error);
+    }
+
+  };
+
+  const handleFormClick = async (e) => {
+    e.preventDefault();
+    if (props.mode == "create"){
+        await postMetric();
+    }
+    else if (props.mode == "edit"){
+        await editMetric(metric_id);
+    }
+    navigate('/dashboard/metrics', {state: { forceRender: true }});
+  }
+
+  let disable = 'false'
+
+  useEffect(()=> {
+    if(props.mode == 'edit'){
+        getMetricById(metric_id)
+        if(props.mode == 'view'){
+            disable = 'true'
+        }
+    }
+  }, [metric_id])
+
+  return (
+    <Layout>
+        <div className="content mt-4 mb-5" style={{width: '100%'}}>
```

Comment:
don't style your component like this unless you really need to use JS to compute your styles.
please remove all the styles added in this way and replace them with classes either from bootstrap or create your own classes and import the styles file.

Reply:
Done